backend/question-service/server/app/readme.md

```diff
@@ -0,0 +1,321 @@
+    - **Query Parameters**:
+        - `difficulty`: List of difficulties to filter by (e.g., `["Easy", "Medium"]`). Optional.
+        - `topic`: List of topics to filter by (e.g., `["Arrays", "Strings"]`). Optional.
+        - **Multiple values can be provided, in which case all questions that satisfy at least one difficulty and one topic will appear.**
+
+    **Sample Request**:
+
+    ```
+    GET /questions # Returns all questions (no filters)
+    ```
+
+    ```
+    GET /questions?difficulty=Easy&topic=Arrays # Returns all Easy questions with topic Arrays
+    ```
+
+    ```
+    GET /questions?difficulty=Easy&difficulty=Medium&topic=Hash%20Tables&topic=Arrays # Returns all Easy or Medium questions with topic Hash Tables or Arrays
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "questions": [
+        {
+            "id": "1",
+            "title": "Find Maximum in Array",
+            ...
+        },
+        {
+            "id": "2",
+            "title": "Reverse Array",
+            ...
+        }
+        ]
+    }
+    }
+    ```
+
+2. `GET /questions/all` Get All Questions
+
+    - **Description**: Returns a list of all questions with all fields.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/all
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "questions": [
+        {
+            "id": "1",
+            "title": "Find Maximum in Array",
+            "difficulty": "Easy",
+            "topic_tags": ["Arrays"],
+            ...
+        },
+        {
+            "id": "2",
+            "title": "Reverse Array",
+            "difficulty": "Easy",
+            "topic_tags": ["Arrays"],
+            ...
+        }
+        ]
+    }
+    }
+    ```
+
+3. `GET /questions/{question_id}` Get a Single Question
+
+    - **Description**: Returns a single question matching the given id.
+    - **Path Parameters**:
+        - `question_id`: The integer ID of the question to retrieve. Required.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/1
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "question": {
+        "id": "1",
+        "title": "Find Maximum in Array",
+        "difficulty": "Easy",
+        "topic_tags": ["Arrays"],
+        ...
+        }
+    }
+    }
+    ```
+
+4. `GET /questions/filters` Get Question Filters
+
+    - **Description**: Returns a list of all possible filters for questions: Difficulty and Topic.
+
+    **Sample Request**:
+
+    ```
+    GET /questions/filters
+    ```
+
+    **Sample Response**:
+
+    ```json
+    {
+    "status": "success",
+    "data": {
+        "Easy": ["Arrays", "Strings"],
+        "Medium": ["Trees", "Graphs"],
+        "Hard": ["Dynamic Programming", "Recursion"]
+    }
+    }
+    ```
+
+### `/admin` Endpoints
+
+1. `POST /admin/questions` Create a New Question
```

Comment:
okay so with this convention, the full endpoint would be
`http://localhost:8003/api/admin/questions` right?

in that case I will make my API gateway proxy requests from
`http://localhost:8001/api/admin/questions` onwards to that endpoint
with admin authorization needed

wdyt?

Reply:
Sounds good to me!